src/profile-logic/stack-timing.js

```diff
@@ -125,8 +125,8 @@ export function getStackTimingByDepth(
   }
 
   // Pop the remaining stacks
-  const endingTime =
-    thread.samples.time[thread.samples.time.length - 1] + interval;
+  const lastIndex = thread.samples.length - 1;
+  const endingTime = thread.samples.time[lastIndex] + interval;
```

Comment:
Here I kept the interval after all (previous versions of the patch used to take the duration), I think it makes more sense after all. Maybe the stack chart could even make sense for a diff thread with this, I'll check it out tomorrow.

Reply:
I believe this will be incorrect for JS tracer information using a duration. The last sample will have a defined duration, and this won't respect it.

Reply:
I think it's easier if you change this in your JS tracer PR, so that you can test with actual data instead of doing it blindly. I hope it's OK for you :)